src/rt/condvar.rs

```diff
@@ -86,11 +86,15 @@ impl Condvar {
 }
 
 impl State {
-    pub(super) fn last_dependent_accesses<'a>(&'a self) -> Box<dyn Iterator<Item = &Access> + 'a> {
-        Box::new(self.last_access.iter())
+    pub(super) fn for_each_last_dependent_access(&self, f: impl FnMut(&Access)) {
```

Comment:
It looks to me that rather than changing this to a `for_each`, this method could just return an `impl Iterator` rather than a `Box`?